### PR TITLE
test: Build rhel image from latest compose

### PR DESCRIPTION
**Pull request by dperpeet**
This uses nightly 7.4 composes. 

 * [ ] FAIL: image-refresh rhel-7-4

### CODE REVIEW

**Comment by dperpeet** [comment]
Subscription failure might be https://bugzilla.redhat.com/show_bug.cgi?id=1426357

**Comment by dperpeet** [comment]
bot: Image refresh for rhel-7

**Comment by cockpituous** [comment]
Image creation for rhel-7 in process on cockpit-11.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-rhel-7-8938b13a5864299db5eb5f2924b24d9af3e64ad5/

**Comment by cockpituous** [comment]
Image creation done: https://github.com/cockpituous/cockpit/commits/refresh-rhel-7-8938b13a5864299db5eb5f2924b24d9af3e64ad5

**Comment by dperpeet** [comment]
new image built to pull in the latest packages from available composes

**Comment by petervo** [comment]
ssh restarting is broken.

**Comment by petervo** [comment]
Do the docker storage tests need updating?

**Comment by dperpeet** [comment]
> Do the docker storage tests need updating?

The compose for 7.4 we're using still has an older version of `atomic`, so I added the relevant known issue. I requested a new compose with the current set of packages.

**Comment by dperpeet** [comment]
I need to generalize the subscriptions test, there is a hardcoded activation key id which I thought was part of the demo data, but it's not static

File: test/verify/check-subscriptions [hunk 1]
```diff
@@ -114,7 +134,7 @@ class TestSubscriptions(MachineCase):
         # enter server and incorrect login data
         b.wait_present("#subscription-register-url-custom")
         b.wait_visible("#subscription-register-url-custom")
-        b.set_val("#subscription-register-url-custom", "{0}:8443/candlepin".format(self.candlepin.address))
+        b.set_val("#subscription-register-url-custom", "{address}:8443/candlepin".format(address=self.candlepin.address))
```
Comment (martinpitt) [inline test/verify/check-subscriptions]: While you are at it, making it simpler instead of longer would just be `self.candlepin.address + ":8843/candlepin"`. (Feel free to ignore, of course, it's correct).

**Comment by dperpeet** [comment]
bot: Image refresh for rhel-7

File: test/verify/check-subscriptions (hunk 1 above)
Comment (dperpeet) [inline test/verify/check-subscriptions]: I don't like to use `+` with strings

**Comment by cockpituous** [comment]
Image creation for rhel-7 in process on cockpit-11.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-rhel-7-dc7b3767f02e2bff39933bd56d318ec8e526d02e/

**Comment by cockpituous** [comment]
Image creation done: https://github.com/cockpituous/cockpit/commits/refresh-rhel-7-dc7b3767f02e2bff39933bd56d318ec8e526d02e

**Comment by dperpeet** [comment]
With the newest compose there seem to be a few issues that require attention.

**Comment by dperpeet** [comment]
blocked on

- [ ] #6401 
- [ ] #6402

**Comment by stefwalter** [comment]
The fact that this is so late is becoming problematic. I'm regenerating the image and going to try to help it get in.

**Comment by stefwalter** [comment]
Rebased and force pushed.

**Comment by stefwalter** [comment]
@dperpeet. Do you think we'll need to keep testing against RHEL 7.3 as well? If so, we'll want to name this image ```rhel-7-4```, right?

**Comment by dperpeet** [comment]
> @dperpeet. Do you think we'll need to keep testing against RHEL 7.3 as well? If so, we'll want to name this image rhel-7-4, right?

We talked about this a while ago and I think it's fine to keep the 7.3 images in the respective branches and this one more forward looking. If necessary we can keep the older one as rhel-7.3

**Comment by stefwalter** [comment]
> We talked about this a while ago and I think it's fine to keep the 7.3 images in the respective branches and this one more forward looking. If necessary we can keep the older one as rhel-7.3

I'm moving all images into bots directory. The end goal is that anything in the respective branches will not be used.

**Comment by dperpeet** [comment]
> I'm moving all images into bots directory. The end goal is that anything in the respective branches will not be used.

In that case I'm for keeping 7.3 named as such and naming this one rhel-7.4 right from the start

**Comment by dperpeet** [comment]
This isn't using the latest Extras yet, but depending on how this goes we may want to add that as a later step.

**Comment by dperpeet** [comment]
bot: Image refresh for rhel-7.4

**Comment by dperpeet** [comment]
bot: Image refresh for rhel-7.4

**Comment by dperpeet** [comment]
docker crash: filed https://bugzilla.redhat.com/show_bug.cgi?id=1451923

**Comment by mvollmer** [comment]
I can't reproduce the `check-networking-bond` failure locally.  These disconnections have happened before.  Maybe it is worth retrying a test that outputs "health check failed".

**Comment by dperpeet** [comment]
Image renamed to `rhel-7-4` to avoid conflicts with filename magic.

File: test/verify/check-docker-storage [hunk 2]
```diff
@@ -65,7 +66,7 @@ def initially_loopbacked(machine):
 # device is added.
 #
 def initially_without_vgroup(machine):
-    return machine.image in [ "fedora-24", "rhel-7" ]
```
Comment (petervo) [inline test/verify/check-docker-storage]: Should we really be removing rhel-7 from this, since this about a new image shouldn't this be left untouched?

File: bots/tests-scan [hunk 3]
```diff
@@ -225,7 +224,7 @@ def scan_for_pull_tasks(api, update, policy):
             # For unmarked and untested status, user must be in whitelist
             # Not this only applies to this specific commit. A new status
             # will apply if the user pushes a new commit.
-            if login not in whitelist:
+            if login not in api.whitelist:
```
Comment (petervo) [inline bots/tests-scan]: Seems ok to be, but is this actually broken now?

File: test/verify/check-docker-storage (hunk 2 above)
Comment (dperpeet) [inline test/verify/check-docker-storage]: good catch, probably a remnant from when I didn't rename the image

File: bots/tests-scan (hunk 3 above)
Comment (dperpeet) [inline bots/tests-scan]: I think it was at some point, but it shouldn't be in this pr either way, definitely not in this commit.

Comment (stefwalter) [inline bots/tests-scan]: Good catch @petervo. ```.whitelist``` is no longer on the GitHub() object on git master.

File: bots/image-scan [hunk 4]
```diff
@@ -61,12 +61,14 @@ DEFAULT_REFRESH = {
...
                       "verify/fedora-25",
-                      "verify/rhel-7" ],
+                      "verify/rhel-7",
+                      "verify/rhel-7-4" ],
         "refresh-days": 30
     },
     "ipa": {
         "triggers": [ "verify/fedora-25",
                       "verify/rhel-7",
+                      "verify/rhel-7-4",
```
Comment (stefwalter) [inline bots/image-scan]: THis also needs to be added to ```REDHAT_REFRESH``` down below.

File: bots/tests-scan [hunk 5]
```diff
@@ -198,7 +198,6 @@ def scan_for_pull_tasks(api, update, policy):
                 if update_status(revision, context, status, changes):
                     results.append(tests_invoke(priority, branch, revision, branch, context))
 
-    whitelist = github.whitelist()
```
Comment (stefwalter) [inline bots/tests-scan]: Seems like this change shouldn't be here.

File: bots/image-scan (hunk 4 above)
Comment (dperpeet) [inline bots/image-scan]: added

File: bots/tests-scan (hunk 5 above)
Comment (dperpeet) [inline bots/tests-scan]: I think all changes in tests-scan were due to a faulty rebase

**Comment by dperpeet** [comment]
rebased because of bots/image-refresh migration

File: bots/tests-scan [hunk 6]
```diff
@@ -49,7 +49,7 @@ DEFAULT_VERIFY = {
 # Non-public images used for testing
 REDHAT_VERIFY = {
     "verify/rhel-7": [ "master", "pulls" ],
-    "verify/rhel-7-4": [ ],
+    "verify/rhel-7-4": [ "pulls" ],
```
Comment (stefwalter) [inline bots/tests-scan]: This needs to say "master", "rhel-7.4", or similar. Ditto for adjoining lines. Should we do this as a follow up?

Comment (dperpeet) [inline bots/tests-scan]: added a commit to change this

File: bots/tests-scan [hunk 7]
```diff
@@ -143,6 +144,12 @@ def prioritize(status, title, labels, priority, context):
     elif state in [ "pending" ]:
         update = None
 
+    # Don't start working on "bot" pull requests automatically.
+    # The bot triggers them explicitly.
+    elif "bot" in labels():
+        priority = None
+        update = None
+
```
Comment (stefwalter) [inline bots/tests-scan]: This reverts the changes from #6845. If you would like to do this, it would make sense to do as a separate pull request with separate discussion.

Comment (dperpeet) [inline bots/tests-scan]: this was a rebase error and unintended

**Comment by dperpeet** [comment]
rebased again

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-7.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-6360-20170614-142222/

**Comment by cockpituous** [comment]
Task failed: http://fedorapeople.org/groups/cockpit/logs/image-refresh-6360-20170614-142222/

**Comment by dperpeet** [comment]
rebuilt with `PackageKit` dependency

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-7.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-6360-20170614-214545/

**Comment by cockpituous** [comment]
Task failed: http://fedorapeople.org/groups/cockpit/logs/image-refresh-6360-20170614-214545/

File: bots/naughty/rhel-7-4/6960-storaged-regression-lvm-resize [hunk 8]
```diff
@@ -0,0 +1,5 @@
+  File "/build/cockpit/bots/../test/verify/check-storage-resize", line *, in testResize
+*
+Error resizing logical volume: Command-line `lvresize "TEST"/"vol" -L 209715200b -r' exited with non-zero exit status 5:
+stdout: `Do you want to unmount "/run/media/admin/FSYS"? [Y|n] n
+*
```
Comment (stefwalter) [inline bots/naughty/rhel-7-4/6960-storaged-regression-lvm-resize]: This is not a known issue with the operating system, unless we file a bug for it. If it's a cockpit thing, we can either workaround the bug in the tests, or fix the bug in Cockpit.

Comment (dperpeet) [inline bots/naughty/rhel-7-4/6960-storaged-regression-lvm-resize]: This might be in cockpit (the commands shouldn't be interactive), but could also be an upstream issue. @mvollmer is currently investigating
I created a known issue for this to get the tests green, but we can also wait for this investigation to result in an upstream bug report or a cockpit fix.

Comment (stefwalter) [inline bots/naughty/rhel-7-4/6960-storaged-regression-lvm-resize]: Thanks for creating a bug.

Comment (dperpeet) [inline bots/naughty/rhel-7-4/6960-storaged-regression-lvm-resize]: For reference, downstream bug: https://bugzilla.redhat.com/show_bug.cgi?id=1462660
